t/01-hash.t

```diff
@@ -56,6 +56,29 @@ is_deeply(
     "slice_exists with default keys",
 );
 
+is_deeply(
+    {slice_delete(\%h, qw(a d))},
+    {
+        b => 2,
+        c => undef
+    },
+    "slice_delete did remove the right keys",
+);
+
+is_deeply(
+    {slice_delete(\%h)},
+    {
+        a => 1,
+        b => 2,
+        c => undef
+    },
+    "slice_delete did not remove anything with default keys",
+);
+
+my %r = slice_delete(\%h, qw(b d));
```

Comment:
What happens to %h?

Reply:
nothing, it keeps the original hash intact and returns a sliced version, just like `slice` itself, it does not modify the original either ... or am i confused ?

Reply:
Then ... slice_delete is probably the wrong name.

Reply:
it probably is the wrong name, checking the previous comment... normal `delete` returns the delete value.

guess it better be named `slice_remove`

Reply:
however, please confirm with me... none of the slicers are modifying the original hash... is it ?

Reply:
Correct, none of them is modifying the original - that's why I would lurk for a name which is not suggesting this one could. Probable such as "slice_neg" (and then it can be reasonable to have a "slice_neg_map" either)